src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -427,7 +429,11 @@ void mf::WlSurface::commit(WlSurfaceState const& state)
     }
     else if (auto const ss = scene_surface())
     {
-        ss.value()->on_next_render(std::move(executor_send_frame_callbacks));
+        std::thread{[callback=std::move(executor_send_frame_callbacks)]()
+            {
+                usleep(10000);
+                callback();
+            }}.detach();
```

Comment:
I guess this is the essence of the proposal. My concerns are twofold:

1. We ought to be scheduling callbacks WRT vsync, not when the frame is requested; and,
2. Spinning up a thread every frame just so it can sleep seems an expensive way to do things.

I realise the vsync information isn't available locally, so I might let that go for a "workaround". But we do have a way (`AlarmFactory`) to schedule callbacks at a future time.

Maybe we can have an executor that in the first iteration runs pending work every interval, and later improve that to sync with vsync?

Reply:
The new thread approach is just for testing, def not intended to be merged. Good idea to abstract the timing code. I will poke around Mir and maybe also look at what wlroots does to see if we can do the right thing.